src/pybind/mgr/cephadm/services/nfs.py

```diff
@@ -55,9 +55,11 @@ def fence_old_ranks(self,
                         self.mgr.spec_store.save_rank_map(spec.service_name(), rank_map)
 
     def config(self, spec: NFSServiceSpec, daemon_id: str) -> None:  # type: ignore
+        from nfs.cluster import create_ganesha_pool
+
         assert self.TYPE == spec.service_type
         assert spec.pool
-        self.mgr._check_pool_exists(spec.pool, spec.service_name())
+        create_ganesha_pool(self.mgr, spec.pool)
```

Comment:
Let's move pool creation to `orch` interface. Then we need not create it in `nfs`, `cephadm` or `rook`.

Reply:
that would require breaking changes in rook, right? In any case what we need is idempotent.